src/NumberGuessingGame.js

```diff
@@ -1,77 +1,79 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import GuessControl from "./GuessControl";
 import GuessMessage from "./GuessMessage";
 import GameOver from "./GameOver";
 
-/**
- *
- * Returns a random integer number from 1-100 inclusive
- */
 function getRandomNumber() {
   return Math.floor(Math.random() * 100) + 1;
 }
 
-const MAX_ATTEMPTS = 5;
+const MAX_ATTEMPTS = 4;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
+export default function NumberGuessingGame  () {
+  const [currentGuess, setCurrentGuess] = useState("");
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
+  const [isCorrectGuess, setIsCorrectGuess] = useState(false);
+  const [isGameOver, setIsGameOver] = useState(false);
+  const [showHint, setShowHint] = useState(false);
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
 
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
-  }
+  const handleGuess = (e) => {
+    e.preventDefault();
+    if (currentGuess === numberToGuess) {
+      setIsGameOver(true);
+      setIsCorrectGuess(true);
+    }
+    else {
+      setShowHint(true);
```

Comment:
Also, I notice the hint goes away while your typing. Maybe keep showing the hint until user guesses again. Then show the diff hint for the current guess. Does that make sense?